### PR TITLE
[web] Migrate Flutter Web DOM usage to JS static interop - 31.

### PR DESCRIPTION
**Pull request by joshualitt**
This is CL 31 in a series of CLs to migrate Flutter Web DOM usage to the new JS static interop API.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 20 new digest(s) on patchset 22.
View them at https://flutter-engine-gold.skia.org/cl/github/33361

File: lib/web_ui/test/engine/semantics/semantics_helper_test.dart [hunk 1]
```diff
@@ -142,26 +141,26 @@ void testMain() {
         expect(mobileSemanticsEnabler.semanticsActivationTimer, isNull);
 
         // Send a click off center
-        _placeholder!.dispatchEvent(html.MouseEvent(
+        _placeholder!.dispatchEvent(createDomMouseEvent(
           'click',
-          clientX: 0,
-          clientY: 0,
+          <Object?, Object?>{ 'clientX': 0,
```
Comment (eyebrowsoffire) [inline lib/web_ui/test/engine/semantics/semantics_helper_test.dart:146]: This spacing is kinda awkward. Could you maybe either do it all on one line, or do like
```
<Object?, Object?>{
  'clientX': 0,
  'clientY": 0,
}
```

File: lib/web_ui/test/engine/semantics/semantics_helper_test.dart [hunk 2]
```diff
@@ -142,26 +141,26 @@ void testMain() {
...
             .toInt();
         final int midY = (activatingElementRect.top +
                 (activatingElementRect.bottom - activatingElementRect.top) / 2)
             .toInt();
-        _placeholder!.dispatchEvent(html.MouseEvent(
+        _placeholder!.dispatchEvent(createDomMouseEvent(
           'click',
-          clientX: midX,
-          clientY: midY,
+          <Object?, Object?>{ 'clientX': midX,
```
Comment (eyebrowsoffire) [inline lib/web_ui/test/engine/semantics/semantics_helper_test.dart:162]: Same here

**Review by eyebrowsoffire — APPROVED** [review]
Just some nitpicks on formatting.

**Pull request merged**